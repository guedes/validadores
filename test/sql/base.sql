\set ECHO 0
BEGIN;
\i sql/validadores.sql
\set ECHO all

SELECT cpf_valido(59328253241);

SELECT 59328253241 #? AS cpf_valido;
SELECT 37821042773 #? AS cpf_valido;
SELECT 91416742433 #? AS cpf_valido;
SELECT 91416000433 #? AS cpf_valido;
SELECT 37821042003 #? AS cpf_valido;
SELECT NOT 37821042003 #? AS cpf_valido;
SELECT NOT 91416000433 #? AS cpf_valido;

CREATE TABLE pessoa (
    nro_cpf cpf
);

INSERT INTO pessoa VALUES(88229346798);
INSERT INTO pessoa VALUES(45476684425);

SAVEPOINT cpf_invalido;
INSERT INTO pessoa VALUES(45076684425);
ROLLBACK TO cpf_invalido;

SAVEPOINT cpf_invalido;
INSERT INTO pessoa VALUES(81249396798);
ROLLBACK TO cpf_invalido;

ROLLBACK;
