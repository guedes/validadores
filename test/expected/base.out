\set ECHO 0
SELECT cpf_valido(59328253241);
 cpf_valido 
------------
 t
(1 row)

SELECT 59328253241 #? AS cpf_valido;
 cpf_valido 
------------
 t
(1 row)

SELECT 37821042773 #? AS cpf_valido;
 cpf_valido 
------------
 t
(1 row)

SELECT 91416742433 #? AS cpf_valido;
 cpf_valido 
------------
 t
(1 row)

SELECT 91416000433 #? AS cpf_valido;
 cpf_valido 
------------
 f
(1 row)

SELECT 37821042003 #? AS cpf_valido;
 cpf_valido 
------------
 f
(1 row)

SELECT NOT 37821042003 #? AS cpf_valido;
 cpf_valido 
------------
 t
(1 row)

SELECT NOT 91416000433 #? AS cpf_valido;
 cpf_valido 
------------
 t
(1 row)

CREATE TABLE pessoa (
    nro_cpf cpf
);
INSERT INTO pessoa VALUES(88229346798);
INSERT INTO pessoa VALUES(45476684425);
SAVEPOINT cpf_invalido;
INSERT INTO pessoa VALUES(45076684425);
ERROR:  value for domain cpf violates check constraint "cpf_check"
ROLLBACK TO cpf_invalido;
SAVEPOINT cpf_invalido;
INSERT INTO pessoa VALUES(81249396798);
ERROR:  value for domain cpf violates check constraint "cpf_check"
ROLLBACK TO cpf_invalido;
ROLLBACK;
